src/Bucket/Bucket.jl

```diff
@@ -0,0 +1,433 @@
+    T0::FT
+    "Conductivity of the soil (W/K/m); constant"
+    κ_soil::FT
+    "Volumetric heat capacity of the soil (J/m^3/K); constant"
+    ρc_soil::FT
+    "Shortwave soil albedo; constant"
+    α_soil::FT
+    "Shortwave snow albedo; constant"
+    α_snow::FT
+    "Critical SWE amount (m) where surface transitions from soil to snow"
+    S_c::FT
+    "Capacity of the land bucket (m)"
+    W_f::FT
+    "Roughness length for momentum (m)"
+    z_0m::FT
+    "Roughness length for scalars (m)"
+    z_0b::FT
+    "Earth Parameter set; physical constants, etc"
+    earth_param_set::PSE
+end
+
+BucketModelParameters(
+    d_soil::FT,
+    T0::FT,
+    κ_soil::FT,
+    ρc_soil::FT,
+    α_soil::FT,
+    α_snow::FT,
+    S_c::FT,
+    W_f::FT,
+    z_0m::FT,
+    z_0b::FT,
+    earth_param_set::PSE,
+) where {FT, PSE} = BucketModelParameters{FT, PSE}(
+    d_soil,
+    T0,
+    κ_soil,
+    ρc_soil,
+    α_soil,
+    α_snow,
+    S_c,
+    W_f,
+    z_0m,
+    z_0b,
+    earth_param_set,
+)
+
+"""
+    PrescribedAtmosphere{FT, LP, TA, UA, QA, RA} <: AbstractAtmosphericDrivers{FT}
+
+Container for holding prescribed atmospheric drivers and other
+information needed for computing turbulent surface fluxes when
+driving the bucket model in standalone mode.
+$(DocStringExtensions.FIELDS)
+"""
+struct PrescribedAtmosphere{FT, LP, TA, UA, QA, RA} <:
+       AbstractAtmosphericDrivers{FT}
+    "Precipitation (m/s) function of time: positive by definition"
+    liquid_precip::LP
+    "Prescribed atmospheric temperature (function of time)  at the reference height (K)"
+    T_atmos::TA
+    "Prescribed wind speed (function of time)  at the reference height (m/s)"
+    u_atmos::UA
+    "Prescribed specific humidity (function of time)  at the reference height (_)"
+    q_atmos::QA
+    "Prescribed air density (function of time)  at the reference height (kg/m^3)"
+    ρ_atmos::RA
+    "Reference height, relative to surface elevation(m)"
+    h_atmos::FT
+    "Surface air density (kg/m^3)"
+    ρ_sfc::FT # Eventually, computed from mean surface pressure (from Atmos) + land T_sfc
+end
+
+function PrescribedAtmosphere(
+    precipitation,
+    T_atmos,
+    u_atmos,
+    q_atmos,
+    ρ_atmos,
+    h_atmos,
+    ρ_sfc,
+)
+    args = (precipitation, T_atmos, u_atmos, q_atmos, ρ_atmos)
+    PrescribedAtmosphere{typeof(h_atmos), typeof.(args)...}(
+        args...,
+        h_atmos,
+        ρ_sfc,
+    )
+end
+
+
+"""
+    CoupledAtmosphere{FT} <: AbstractRadiativeDrivers{FT}
+
+To be used when coupling to an atmosphere model; internally, used for
+multiple dispatch on `surface_fluxes`.
+"""
+struct CoupledAtmosphere{FT} <: AbstractRadiativeDrivers{FT} end
+
+
+"""
+    PrescribedRadiation{FT, SW, LW} <: AbstractRadiativeDrivers{FT}
+
+Container for the prescribed radiation functions needed to drive the
+bucket model in standalone mode.
+$(DocStringExtensions.FIELDS)
+"""
+struct PrescribedRadiation{FT, SW, LW} <: AbstractRadiativeDrivers{FT}
```

Comment:
Would `PrescribedRadiationSurfaceFluxes` be a better name or is this obvious to a "land person"? 🤔  Also, for idealized testing in land, would it be useful to have `PrescribedTurbulentSurfaceFluxes` (i.e. bypass `SurfaceFluxes.jl` for LHF and SHF)?

Reply:
I changed the name to PrescribedRadiativeFluxes - good suggestion. I think for now we can hold off on adding in a PrescribedTurbulentSurfaceFluxes type (unless there is a use case for it at the moment). Since we have the `AbstractAtmosphericDrivers` and `AbstractRadiativeDrivers` types set up, and two example of each, we are ready to add in when needed.